singlestoredb/fusion/handlers/job.py

```diff
@@ -0,0 +1,642 @@
+
+
+class RunJobHandler(SQLHandler):
+    """
+    RUN JOB USING NOTEBOOK notebook_path
+        [ with_runtime ]
+        [ with_parameters ]
+    ;
+
+    # Path to notebook file
+    notebook_path = '<notebook-path>'
+
+    # Runtime to use
+    with_runtime = WITH RUNTIME '<runtime-name>'
+
+    # Parameters to pass to the job
+    with_parameters = WITH PARAMETERS <json>
+
+    Description
+    -----------
+    Creates a scheduled notebook job that runs once immediately on the specified runtime.
+
+    Arguments
+    ---------
+    * ``<notebook-path>``: The path in the Stage where the notebook file is stored.
+    * ``<runtime-name>``: The name of the runtime the job will be run with.
+    * ``<json>``: The parameters to pass to the job. A JSON object with
+      the following format: ``{"<paramName>": "<paramValue>", ...}``.
+
+    Remarks
+    -------
+    * The job is run immediately after the command is executed.
+    * The ``WITH RUNTIME`` clause specifies the name of the runtime that
+      the job will be run with.
+    * The ``WITH PARAMETERS`` clause specifies the parameters to pass to the job. The
+      only supported parameter value types are strings, integers, floats, and booleans.
+
+    Example
+    -------
+    The following command creates a job that will run the content of notebook
+    **example_notebook.ipynb** using the runtime **notebooks-cpu-small** immediately.
+    The job will have the following parameters: **strParam** with value **"string"**,
+    **intParam** with value **1**, **floatParam** with value **1.0**, and **boolParam**
+    with value **true**::
+
+        RUN JOB USING NOTEBOOK 'example_notebook.ipynb'
+           WITH RUNTIME 'notebooks-cpu-small'
+           WITH PARAMETERS {
+                              "strParam": "string",
+                              "intParam": 1,
+                              "floatParam": 1.0,
+                              "boolParam": true
+                            }
+        ;
+
+    """
+
+    def run(self, params: Dict[str, Any]) -> Optional[FusionSQLResult]:
+        res = FusionSQLResult()
+        res.add_field('JobID', result.STRING)
+
+        jobs_manager = get_workspace_manager().organizations.current.jobs
+
+        parameters = None
+        if params.get('with_parameters'):
+            parameters = {}
+            for name, value in params['with_parameters'].items():
+                parameters[name] = value
+
+        job = jobs_manager.run(
+            params['notebook_path'],
+            runtime_name=params['with_runtime'],
+            parameters=parameters,
+        )
+        res.set_rows([(job.job_id,)])
+
+        return res
+
+
+RunJobHandler.register(overwrite=True)
+
+
+class WaitOnJobsHandler(SQLHandler):
+    """
+    WAIT ON JOBS job_ids
+        [ with_timeout ]
+    ;
+
+    # Job IDs to wait on
+    job_ids = '<job-id>',...
+
+    # Timeout in seconds
+    with_timeout = WITH TIMEOUT <integer>
```

Comment:
I'm wondering if we should have time units here too. I don't know what the maximum timeout could be, but I could see people wanting minutes here as well, maybe even hours.

Reply:
Done